tidy3d/components/heat_charge/monitor_data.py

```diff
@@ -235,4 +242,220 @@ def symmetry_expanded_copy(self) -> VoltageData:
         return self.updated_copy(voltage=new_phi, symmetry=(0, 0, 0))
 
 
-HeatChargeMonitorDataType = Union[TemperatureData, VoltageData]
+class HeatChargeDataset(Tidy3dBaseModel):
+    """Class that deals with parameter-depending fields."""
+
+    base_data: FieldDataset = pd.Field(title="Base data", description="Spatial dataset")
+
+    field_series: Dict[str, Tuple[IndexedDataArray, ...]] = pd.Field(
+        title="Field series", description="Dictionary of field solutions. "
+    )
+
+    doping: Dict[str, IndexedDataArray] = pd.Field(
+        title="Doping", description="Dictionary containing the doping distributions."
+    )
+
+    parameter_array: Tuple[pd.FiniteFloat, ...] = pd.Field(
+        title="Parameter array",
+        description="Array containing the parameter values at which the field series are stored.",
+    )
+
+    @cached_property
+    def num_fields_saved(self):
+        """Number of fields stored"""
+        return len(self.parameter_array)
+
+    def get_field(self, field: str, loc: int):
+        """Returns the field specified by 'field' stored at the position specified by 'loc'"""
+
+        assert loc < self.num_fields_saved
+        return self.base_data.updated_copy(values=self.field_series[field][loc])
+
+    def get_doping(self, doping_type: str = "Acceptors"):
+        """Returns the doping specified by 'doping_type'. This, can take the values:
+        'Acceptors' or 'Donors'"""
+
+        return self.base_data.updated_copy(values=self.doping[doping_type])
+
+    @pd.root_validator(skip_on_failure=True)
+    def check_data_and_params_have_same_length(cls, values):
+        """Check that both field series and parameter array have the same length."""
+
+        field_series = values["field_series"]
+        parameter_array = values["parameter_array"]
+        for key in field_series.keys():
+            assert len(field_series[key]) == len(parameter_array)
+
+        return values
+
+
+class ChargeSimulationData(HeatChargeMonitorData):
+    """Class that stores Charge simulation data.
+    Example
+    -------
+    ...
+    """
+
+    monitor: Union[ChargeSimulationMonitor] = pd.Field(
+        ..., title="Monitor", description="Data associated with a Charge simulation."
+    )
+
+    data_series: Optional[HeatChargeDataset] = pd.Field(
+        None, title="Data series", description="Contains the data."
+    )
+
+    capacitance_curve: Optional[Dict[str, CapacitanceCurveDataArray]] = pd.Field(
+        None,
+        title="Capacitance curve",
+        description="Small signal capacitance associated to the monitor.",
+    )
```

Comment:
So I feel this is related to some of the previous comments. Like ultimately we're implementing nonlinear interdependent parametric functionality that represents the physical relationships between capacitance, charge, doping etc. So I feel we need a more generic way of representing these data/physical relationships than indeependent from their final plot. 

Like if this were math or at least in the models normally it can implement relationships like C(N_a, V) basically or V(t, C). So how do we abstract these parametric relationships in a way that is complete and application independent

Reply:
Hmm maybe this is why it is worth using specific parameter naming, because if we do implement a more general class in the future that needs to have the data restructured to account for the full physics it might be easier.  This is my personal opinion though. If you think it'll be easier to generalise eventually for the same name without making the class incompatible with the past then also open for a more generic name - just trying to visualise how it might look and think about the foundations solidly